ydb/core/blobstorage/vdisk/balance/handoff_map.h

```diff
@@ -202,117 +174,110 @@ namespace NKikimr {
-                        Y_ABORT("Implement"); // FIXME
-                    } else if (memRec->GetType() == TBlobType::DiskBlob) {
-                        const auto& merger = dataMerger->GetDiskBlobMerger();
-                        const TDiskBlob& blob = merger.GetDiskBlob();
-                        ui32 fullDataSize = blob.GetFullDataSize();
-
-                        // iterate via parts
-                        for (TDiskBlob::TPartIterator it = blob.begin(), e = blob.end(); it != e; ++it) {
-                            ui8 partId = it.GetPartId();
-                            Y_DEBUG_ABORT_UNLESS(partId > 0);
-                            if (moveVec.Get(partId - 1)) {
-                                TLogoBlobID id(key.LogoBlobID(), partId);
-                                TVDiskIdShort vdisk(memRec->GetIngress().GetMainReplica(Top.get(), id));
-                                // FIXME: for huge blobs we need to implement an actor which makes read, because we don't have
-                                // this blob in memory
-                                TRope temp;
-                                bool res = HandoffDelegate->Restore(ctx, vdisk, id, fullDataSize, it.GetPart());
-                                Stat.SuccessfulMoveSend += ui64(res);
-                                Stat.FailedMoveSend += ui64(!res);
-                            }
-                        }
-                    } else
-                        Y_ABORT("Unexpected case");
+            return TrRes.SetRmData(key, memRec, dataMerger);
+        }
+
+        const TTransformedItem *defaultResult = TrRes.SetRaw(key, memRec, dataMerger); // unchanged by default
+        if (!RunHandoff) {
+            return defaultResult;
+        }
+
+        Y_DEBUG_ABORT_UNLESS(Counter < DelMap.size());
+        const NMatrix::TVectorType localVec = memRec->GetIngress().LocalParts(Top->GType);
+        ui8 vecSize = Top->GType.TotalPartCount();
+
+        const NMatrix::TVectorType delPlan(DelMap.at(Counter), vecSize);
+        const NMatrix::TVectorType delVec = delPlan & localVec;
+
+        if (delVec.Empty()) {
+            return defaultResult;
+        }
+
+        TIngress ingress = memRec->GetIngress(); // ingress we are going to change
+        NMatrix::TVectorType localParts = ingress.LocalParts(Top->GType);
+
+        switch (memRec->GetType()) {
+            case TBlobType::DiskBlob: {
+                const TDiskBlob& blob = dataMerger->GetDiskBlobMerger().GetDiskBlob();
+
+                // mark deleted handoff parts in ingress
+                for (ui8 i = localParts.FirstPosition(); i != localParts.GetSize(); i = localParts.NextPosition(i)) {
+                    if (delVec.Get(i)) {
+                        const ui8 partId = i + 1;
+                        TLogoBlobID id(key.LogoBlobID(), partId);
+                        ingress.DeleteHandoff(Top.get(), HullCtx->VCtx->ShortSelfVDisk, id, true);
```

Comment:
DeleteHandoff() sets the deletion bit in handoff vector, which is already set, 'cause it is in delVec
this statement seems useless